### DISCUSSION TITLE
System error 5 has occurred.  Access is denied.

### DISCUSSION THREAD

**Discussion by aflame13** [crowbarmaster/BedrockManagementService#80] [Q&A] [answered]
Hello! I run 3 Bedrock servers and struggle with the time it takes to update and maintain them. I found this project and it looks amazing! But I am unfortunately not a tech wizard and I ran into an issue. I did some googling to figure out my problem but can't, so I figured I'd post here.

As I understand it, the BedrockService can run automatically with Windows if it is installed as a service. I tried following the quickstart instructions and when I try to start the service in Admin command prompt, I get the error...

"System error 5 has occurred.

Access is denied."

I'm confused because I think the files have the correct permissions and I am signed in as an Admin, using the Admin cmd prompt. Any ideas?

Edit: I am running this on an up to date Windows 10 computer. Also, I can run the service .exe manually and it works. I've been able to set up my 3 servers in BedrockManager and they run perfectly. I'd just like to set it up as a Windows service so it automatically runs when the server starts, unless this isn't a feature and I'm misunderstanding.

**Comment by crowbarmaster** [comment, marked as answer]
My apologies on not getting back to this sooner, I got carried away and forgot. Anyways, what path are you running the service out of? If its a protected folder such as "C:\Users\YourUsername\BMS", this would be a result. The problem is when running as a service, the user it runs under is "LocalService". It would seem that possibly for security reasons, LocalService is denied access to this path. Try moving it to a folder outside user or system folders, and get back to me!

**Reply by aflame13** [reply]
Thank you for your reply! I appreciate the work you’ve put into this project, it’s awesome.

I tried moving the folder but now when I try to start the service in Admin Cmd prompt from the correct folder, it gives me the error “System error 2 has occurred. The system cannot find the file specified.” Maybe running it and trying to install from the desktop (the folder I had it in originally) messed it up? Should I try reinstalling?

If I should reinstalling, how would I uninstall it? I could delete the folder but the service would still be installed eh?

**Reply by crowbarmaster** [reply]
Ah so your only issue here is that the service details point to the old location. Toss BedrockService.Service.exe back in the old folder, run the uninstall command via admin prompt, then install the service from the new location. This should fix you up!

**Reply by aflame13** [reply]
Thank you again! I manually updated the servers so not sure the auto update works, but I figure it does as the service is now working properly now that I moved it off the desktop.

I’m not sure if there’s a place for ideas, but I’ll leave one here. I see you’ve worked on implementing a Pack Manager which is very cool. I have a modded server and it’s a pain any time a mod updates and I have to go to the server, swap out the folders, and manually update the pack IDs. If you can get the R/B Pack Manager working, do you think it’d be possible to be able to make a way to easily update packs? My idea is you’d set it up so you can select a pack folder on the server from the left list, then select the corresponding updated pack folder on the client computer from the right list, click an “update addon” button, and it would replace the pack on the server with the one from your client while also automatically updating the pack ID in the json world pack list. That would be amazing.

Lastly, is there any way to send you money? I’m not rich but I appreciate your work and would send you a thank you.
